runtime/drivers/clickhouse/olap.go

```diff
@@ -659,6 +728,46 @@ func (c *connection) acquireConn(ctx context.Context) (*sqlx.Conn, func() error,
 	return conn, release, nil
 }
 
+func (c *connection) getTableEngine(ctx context.Context, name string) (string, error) {
+	var engine string
+	res, err := c.Execute(ctx, &drivers.Statement{
+		Query:    "SELECT engine FROM system.tables WHERE database = currentDatabase() AND name = ?",
+		Args:     []any{name},
+		Priority: 1,
+	})
+	if err != nil {
+		return "", err
+	}
+	defer res.Close()
+	if res.Next() {
+		if err := res.Scan(&engine); err != nil {
+			return "", err
+		}
+	}
+	return engine, nil
+}
+
+func (c *connection) getTablePartitions(ctx context.Context, name string) ([]string, error) {
+	res, err := c.Execute(ctx, &drivers.Statement{
```

Comment:
Pass it as `arg` similar to function above(preferable)  or wrap name in `safeSQLName`.